components/ProfileMenu.vue

```diff
@@ -0,0 +1,119 @@
+<script setup>
+import { Menu, MenuButton, MenuItems, MenuItem } from "@headlessui/vue";
+</script>
+
+<template>
+  <div class="text-right">
+    <Menu as="div" class="relative inline-block text-left">
+      <div>
+        <MenuButton
+          class="flex w-full justify-center rounded-md bg-opacity-20 px-4 text-black"
+        >
+          Profile
+          <IconComponent name="nav.profile" class="ml-2 -mr-1 h-5 w-5" />
+        </MenuButton>
+      </div>
+
+      <transition
+        enter-active-class="transition duration-100 ease-out"
+        enter-from-class="transform scale-95 opacity-0"
+        enter-to-class="transform scale-100 opacity-100"
+        leave-active-class="transition duration-75 ease-in"
+        leave-from-class="transform scale-100 opacity-100"
+        leave-to-class="transform scale-95 opacity-0"
+      >
+        <MenuItems
+          class="absolute right-0 mt-2 w-56 origin-top-right divide-y divide-gray-100 rounded-md bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none"
+        >
+          <div class="px-1 py-1">
+            <MenuItem v-slot="{ active }">
```

Comment:
All those menu items seem to have the same property. Can't we use a loop here..?

I'd only include options which are done yet or show some kind of message - not something which we are about to implement.

Reply:
I just look at the headlessui documentation. I would like also to refact it to have less code. Would you let me know how to do it ? 
-------------------
Also, i've installed nuxt-headlessui module and set it up in nuxt.config.ts file. But i am still getting red line when i am trying to import elements i need for the menu